tools/walletextension/test/wallet_extension_test.go

```diff
@@ -216,26 +218,35 @@ func TestCanSubscribeForLogsOverWebsockets(t *testing.T) {
 	defer timeout.Stop()
 
 	// We watch the connection to receive a log...
-	_, respJSON, err := conn.ReadMessage()
+	_, logRespJSON, err := conn.ReadMessage()
 	if err != nil {
 		t.Fatalf("could not read log from websocket. Cause: %s", err)
 	}
 
-	var resp map[string]interface{}
-	err = json.Unmarshal(respJSON, &resp)
+	var logResp map[string]interface{}
+	err = json.Unmarshal(logRespJSON, &logResp)
 	if err != nil {
 		t.Fatalf("could not unmarshal received log from JSON")
 	}
 
 	// We extract the topic from the received logs. The API should have set this based on the filter we passed when subscribing.
-	logMap := resp[common.JSONKeyParams].(map[string]interface{})[common.JSONKeyResult].(map[string]interface{})
+	logMap := logResp[common.JSONKeyParams].(map[string]interface{})[common.JSONKeyResult].(map[string]interface{})
 	logTopic := logMap[jsonKeyTopics].([]interface{})[0].(string)
 
 	if !strings.Contains(logTopic, dummyHash.Hex()) {
 		t.Fatalf("expected response containing '%s', got '%s'", dummyHash.Hex(), logTopic)
 	}
 }
 
+// Checks that the response to a subscription request is correctly-formatted.
+func validateSubscriptionResponse(t *testing.T, resp []byte) {
```

Comment:
I use a regex to ensure the response is well-formed. I should probably do this for the other tests too. I've opened a ticket.

Reply:
It might be cleaner/less fragile to just use the json parser. Formatting and ordering shouldn't matter and it can fish out the values as part of that.